app/src/main/java/com/nervousfish/nervousfish/data_objects/VerificationMethod.java

```diff
@@ -19,6 +19,9 @@
      * @param verificationMethodEnum The verification method this object should represent.
      */
     public VerificationMethod(final VerificationMethodEnum verificationMethodEnum) {
+        if (verificationMethodEnum == null) {
+            throw new IllegalArgumentException("The verification method may not be null");
+        }
         this.verificationMethodEnum = verificationMethodEnum;
```

Comment:
You didn't touch it, but can you rename this attribute to `verificationMethod` 😇